### PR TITLE
Metrics Refactor and record metrics around L1 Server-Timings

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Some refactor/fixes of the metrics code for correctness/grokability/dryness
Add more dimensions/labels to metrics to better capture block/car, cache-hit/cache-miss  cases
Record prometheus metrics around important L1 server-timings

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -35,101 +44,136 @@ var (
...
+		Help:    "Latency observed during failed caboose fetches from a single peer in milliseconds",
+		Buckets: durationMsPerBlockHistogram,
 	})
 
-	fetchDurationPerCarPerPeerFailureMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
-		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_car_peer_failure"),
-		Help:    "Latency observed during failed caboose car fetches from a single peer",
-		Buckets: durationPerCarHistogram,
+	fetchDurationBlockFailureMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_block_failure"),
```
Comment (willscott) [inline metrics.go:106]: renaming current metrics will break current dashboards - why is this changing?

Comment (aarshkshah1992) [reply metrics.go:106]: @willscott Not all the metrics have been renamed. Only some that were either incorrectly named or some that have taken the place of some repetitive ones.

I'll ensure the Caboose/Bifrost dashboard/metrics encompass the correct names. We're anyways revamping both the dashboards.

File: fetcher.go [hunk 2]
```diff
@@ -284,6 +284,11 @@ func updateSuccessServerTimingMetrics(timingHeaders []string, resourceType strin
 					fetchDurationPerPeerSuccessCacheMissTotalLassieMetric.WithLabelValues(resourceType).Observe(float64(m.Duration.Milliseconds()))
 				case "nginx":
 					fetchDurationPerPeerSuccessTotalL1NodeMetric.WithLabelValues(resourceType, cache_status).Observe(float64(m.Duration.Milliseconds()))
+					networkTimeMs := totalTimeMs - m.Duration.Milliseconds()
+					fetchNetworkSpeedPerPeerSuccessMetric.WithLabelValues(resourceType).Observe(float64(recieved) / float64(networkTimeMs))
```
Comment (willscott) [inline fetcher.go:288]: what about when `networkTimeMs` is 0? - that leads to a division by 0

**Review by willscott — APPROVED** [review]

File: fetcher.go (hunk 2 above)
Comment (aarshkshah1992) [reply fetcher.go]: Fixed and added more sanity checks.

**Pull request merged**